src/ACFFields.php

```diff
@@ -0,0 +1,153 @@
+<?php
+/**
+ * PHP version 5.6
+ *
+ * @package wpml-to-polylang
+ */
+
+namespace WP_Syntex\WPML_To_Polylang;
+
+defined( 'ABSPATH' ) || exit;
+
+/**
+ * Handles the ACF fields.
+ *
+ * @since 0.6
+ */
+class ACFFields extends AbstractSteppable {
+	const ACFML_KEY = 'wpml_cf_preferences';
+	const TYPES     = [
+		0 => 'none', // WPML_IGNORE_CUSTOM_FIELD.
+		1 => 'copy', // WPML_COPY_CUSTOM_FIELD.
+		2 => 'translate', // WPML_TRANSLATE_CUSTOM_FIELD.
+		3 => 'copy_once', // WPML_COPY_ONCE_CUSTOM_FIELD.
+	];
+
+	/**
+	 * Batch size.
+	 *
+	 * @var int
+	 */
+	protected $batch_size;
+
+	/**
+	 * Constructor.
+	 *
+	 * @since 0.6
+	 */
+	public function __construct() {
+		$this->batch_size = absint( WPML_TO_POLYLANG_QUERY_BATCH_SIZE / 100 ); // 50 by default, to limit the size of the UPDATE query.
+	}
+
+	/**
+	 * Returns the action name.
+	 *
+	 * @since 0.6
+	 *
+	 * @return string
+	 */
+	public function getName() {
+		return 'process_acf_fields';
+	}
+
+	/**
+	 * Returns the processing message.
+	 *
+	 * @since 0.6
+	 *
+	 * @return string
+	 */
+	protected function getMessage() {
+		return esc_html__( 'Processing ACF fields', 'wpml-to-polylang' );
+	}
+
+	/**
+	 * Processes the ACF fields.
+	 *
+	 * @since 0.6
+	 *
+	 * @return void
+	 */
+	protected function handle() {
+		global $wpdb;
+
+		$offset  = absint( ( $this->step * $this->batch_size ) - $this->batch_size );
```

Comment:
Isn't it necessary to override `getPercentage()` as you change the size?

Or maybe better: introduce a method `getBatchSize()` in `AbstractSteppable` overriden in this class?

Reply:
HA! I totally missed it!